pkg/shp/zipreader_test.go

```diff
@@ -0,0 +1,87 @@
+package shp
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestReadZipFrom(t *testing.T) {
+
+	p := "test_files/ne_110m_admin_0_countries.zip"
+
+	ior, err := os.Open(p)
+	assert.Nil(t, err)
+	defer func(ior *os.File) {
+		err := ior.Close()
+		assert.Nil(t, err)
+	}(ior)
+
+	zr, err := ReadZipFrom(ior)
+	assert.Nil(t, err)
+	defer func(zr *ZipReader) {
+		err := zr.Close()
+		assert.Nil(t, err)
+	}(zr)
+
+	var shps []Shape
+	for zr.Next() {
+		_, shp := zr.Shape()
+		shps = append(shps, shp)
+	}
+	assert.Nil(t, zr.Err())
+	assert.Equal(t, 177, len(shps))
+}
+
+func TestReadZipFromWrongScenarios(t *testing.T) {
+	t.Parallel()
+	testCases := []struct {
+		name  string
+		input string
+	}{
+		{
+			name:  "ReadZipFromWrongFile",
+			input: "test_files/point.shp",
+		},
+		{
+			name:  "ReadZipFromEmptyZip",
+			input: "test_files/empty.zip",
+		},
+		{
+			name:  "ReadZipFromMultiZip",
+			input: "test_files/multi.zip",
+		},
+	}
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(tt *testing.T) {
+			tt.Parallel()
+
+			ior, err := os.Open(tc.input)
+			assert.Nil(t, err)
+			defer func(ior *os.File) {
+				err := ior.Close()
+				assert.Nil(t, err)
+			}(ior)
+
+			_, err = ReadZipFrom(ior)
+			assert.NotNil(tt, err)
+			// assert.True(tt, errors.As(tc.expected,$err))
```

Comment:
If you do not plan to use this line, please delete it. If not, please write a comment with the reason.